File: beam-ocaml/lib/dispatch.ml
(* responsible for reading messages and dispatching to the appropriate function *)

type cmd =
  | READ_DIR of string
  | READ_FILE of string
  | WRITE_FILE of string * string
[@@deriving yojson, show]

let call = function
  | READ_DIR dirname -> Ops.read_dir dirname
  | READ_FILE filename -> Ops.read_file filename
  | WRITE_FILE (filename, contents) -> Ops.write_file filename contents
;;

File: beam-py/dispatch.py
import ops

dispatch_table = {
    'read_file': ops.read_file,
    'write_file': ops.write_file,
    'read_dir': ops.read_dir,
}

def run(cmd, args):
    if cmd in dispatch_table:
        return dispatch_table[cmd](*args)
    else:
        return (False, f'Command not found: {cmd}')
